independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -119,6 +127,30 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
         return bean;
     }
 
+    private static ScopeInfo inheritScope(ClassInfo beanClass, BeanDeployment beanDeployment) {
+        DotName superClassName = beanClass.superName();
+        while (!superClassName.equals(DotNames.OBJECT)) {
+            ClassInfo classFromIndex = beanDeployment.getIndex().getClassByName(superClassName);
+            if (classFromIndex == null) {
+                // class not in index
+                LOGGER.logf(Logger.Level.WARN, "Arc is not able to determine scope for bean " + beanClass
+                        + " using inheritance because its super class " + superClassName
+                        + "is not part of Jandex index. Dependent scope will be used instead.");
+                return null;
+            }
+            for (AnnotationInstance annotation : classFromIndex.classAnnotations()) {
+                ScopeInfo scopeAnnotation = beanDeployment.getScope(annotation.name());
```

Comment:
So you assume all scope annotations declare `@Inherited`...

Reply:
Yes I do, which is incorrect assuming custom ones are in play.
Thanks for pointing that out.